_posts/2023-03-22-dd_00005_half_adder.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: single
+title: Combinatorial Circuits - The Half Adder
+category: Digital Design
+permalink: /categories/digital_design/half_adder
+excerpt: "How to add 1-bit numbers in hardware."
+description:
+  "This post introduces the Half-Adder, a circuit which forms the basis of any
+  digital addition circuitry."
+author: Kirthihan Yasotharan
+toc: false
+toc_label: "Content"
+toc_sticky: true
+toc_icon: "align-justify"
+header:
+  overlay_color: "#000"
+  overlay_filter: "0.5"
+  overlay_image: /assets/images/digital_design.jpg
+---
+
+In our last two posts we have learned what logic gates are, which kind of gates
+exist and how they can be implemented using transistors.
+
+Now it's time to see what all stuff we can do with these things! Have you ever
+wondered why a computer is called "computer"? Well, as the name suggests,
+computers calculate, or _compute_, stuff. What and how they do it is something
+we will cover in the future, but for now we'll focus on the essentials.
+
+It turns out that we can use logic gates to do all kinds of **mathematical
+operations**. In this post, we will have a look at the most basic of
+mathematical operations: the **addition**. How can we add two numbers using just
+simple logic gates, you ask?
+
+Well, for that we need some basic understanding of **binary numbers** and how
+addition works in the binary system. As this blog series focuses more on the
+hardware side, we assume that you are already familiar with **binary numbers**.
+If not, or you need to refresh your memory, then feel free to consult a search
+engine and/or latest AI of your choice to learn the basics about binary numbers.
+We recommend giving the
+[Wikipedia article](https://en.wikipedia.org/wiki/Binary_number) about binary
+numbers a read. Of course, there are also plenty of tutorials on YouTube, such
+as [this](https://www.youtube.com/watch?v=LpuPe81bc2w) one on the basics of
+binary numbers, followed by [this](https://www.youtube.com/watch?v=C5EkxfNEMjE)
+one on doing addition in the binary system. We might also cover binary numbers
+in a future post in our _Getting Started_ category, so stay tuned for that. Ok,
```

Comment:
maybe a link to getting started category?

Reply:
Good point -> fixed: [a48611c](https://github.com/bitsandvolts/bitsandvolts.github.io/pull/49/commits/a48611c7f41043e744373327dc0e13fdf157dbe0)